extension/test/test-logger.js

```diff
@@ -10,7 +10,13 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
   var message = '';
 
   return function (status) {
-    var line = {}, hasMore;
+    var line = {}, hasMore, isSuccessStatus;
+
+    isSuccessStatus = typeof status == "number";
+    test.assert(isSuccessStatus, status);
+    if (!isSuccessStatus) {
+      return;
+    }
```

Comment:
> [...] I have to admit that I don't know anymore what the status parameter is here. [...]

`status` parameter could be an error parameter of an `onRejected` handler or a number parameter (written bytes) of an `onFullfilled` handler from a `write` promise.